ui/src/actions/remoteAccess.js

```diff
@@ -41,57 +36,51 @@ export function requestControl(
   name = '',
   userInfo = {},
 ) {
-  return () => {
-    sendRequestControl(control, message, name, userInfo);
-  };
+  return () => sendRequestControl(control, message, name, userInfo);
 }
 
 export function takeControl() {
-  return (dispatch) => {
-    sendTakeControl().then(() => {
-      dispatch(getLoginInfo());
-      dispatch(getRaState());
-    });
+  return async (dispatch) => {
+    await sendTakeControl();
+    dispatch(getLoginInfo());
+    dispatch(getRaState());
   };
 }
 
 export function giveControl(username) {
-  return (dispatch) => {
-    sendGiveControl(username).then(() => {
-      dispatch(getLoginInfo());
-      dispatch(getRaState());
-    });
+  return async (dispatch) => {
+    await sendGiveControl(username);
+    dispatch(getLoginInfo());
+    dispatch(getRaState());
   };
 }
 
 export function logoutUser(username) {
-  return (dispatch) => {
-    sendLogoutUser(username).then(() => {
-      dispatch(getLoginInfo());
-      dispatch(getRaState());
-    });
+  return async (dispatch) => {
+    await sendLogoutUser(username);
+    dispatch(getLoginInfo());
+    dispatch(getRaState());
   };
 }
 
 export function respondToControlRequest(giveControl = true, message = '') {
-  return (dispatch) => {
-    sendRespondToControlRequest(giveControl, message).then(() => {
-      dispatch(getLoginInfo());
-      dispatch(getRaState());
-    });
+  return async (dispatch) => {
+    await sendRespondToControlRequest(giveControl, message);
+    dispatch(getLoginInfo());
+    dispatch(getRaState());
   };
 }
 
 export function updateAllowRemote(allow) {
-  return (dispatch) => {
-    sendUpdateAllowRemote(allow);
+  return async (dispatch) => {
+    await sendUpdateAllowRemote(allow);
     dispatch({ type: 'SET_ALLOW_REMOTE', allow });
   };
 }
 
 export function updateTimeoutGivesControl(timeoutGivesControl) {
-  return (dispatch) => {
-    sendUpdateTimeoutGivesControl(timeoutGivesControl);
+  return async (dispatch) => {
+    await sendUpdateTimeoutGivesControl(timeoutGivesControl);
     dispatch({ type: 'SET_TIMEOUT_GIVES_CONTROL', timeoutGivesControl });
   };
```

Comment:
Notice that there were no `then` chains, so the store was updated "optimistically". This was risky since there is no error handling, so the store could get out of sync. I'm now waiting for the request to succeed before updating the store. This may make the checkboxes less responsive, but at least it is safer for now.